[bug] Timeout on data update

Retrying doesn't always help. I should set a [`backoff_factor`](https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html#urllib3.util.Retry) on the `Retry`.

```
INFO:fx:downloading MUFG quotes for currency pair USD/JPY for 2025-03-04 to 2025-03-18...
WARNING:urllib3.connectionpool:Retrying (Retry(total=9, connect=4, read=5, redirect=2, status=None)) after connection broken by 'ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f999abe6cb0>, 'Connection to murc-kawasesouba.jp timed out. (connect timeout=10.0)')': /fx/past_3month_result.php?y=2025&m=03&d=12
WARNING:urllib3.connectionpool:Retrying (Retry(total=9, connect=4, read=5, redirect=2, status=None)) after connection broken by 'ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f999a998c70>, 'Connection to murc-kawasesouba.jp timed out. (connect timeout=10.0)')': /fx/past_3month_result.php?y=2025&m=03&d=15
WARNING:urllib3.connectionpool:Retrying (Retry(total=9, connect=4, read=5, redirect=2, status=None)) after connection broken by 'ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f999a99aec0>, 'Connection to murc-kawasesouba.jp timed out. (connect timeout=10.0)')': /fx/past_3month_result.php?y=2025&m=03&d=16
WARNING:urllib3.connectionpool:Retrying (Retry(total=8, connect=3, read=5, redirect=2, status=None)) after connection broken by 'ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f999a9987c0>, 'Connection to murc-kawasesouba.jp timed out. (connect timeout=10.0)')': /fx/past_3month_result.php?y=2025&m=03&d=16
WARNING:urllib3.connectionpool:Retrying (Retry(total=7, connect=2, read=5, redirect=2, status=None)) after connection broken by 'ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f999aaa9750>, 'Connection to murc-kawasesouba.jp timed out. (connect timeout=10.0)')': /fx/past_3month_result.php?y=2025&m=03&d=16
WARNING:urllib3.connectionpool:Retrying (Retry(total=6, connect=1, read=5, redirect=2, status=None)) after connection broken by 'ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f999aaa9870>, 'Connection to murc-kawasesouba.jp timed out. (connect timeout=10.0)')': /fx/past_3month_result.php?y=2025&m=03&d=16
WARNING:urllib3.connectionpool:Retrying (Retry(total=5, connect=0, read=5, redirect=2, status=None)) after connection broken by 'ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f999aaa9990>, 'Connection to murc-kawasesouba.jp timed out. (connect timeout=10.0)')': /fx/past_3month_result.php?y=2025&m=03&d=16
Traceback (most recent call last):
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/connection.py", line 198, in _new_conn
    sock = connection.create_connection(
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/util/connection.py", line 85, in create_connection
    raise err
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/util/connection.py", line 73, in create_connection
    sock.connect(sa)
TimeoutError: timed out

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 787, in urlopen
    response = self._make_request(
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 488, in _make_request
    raise new_e
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 464, in _make_request
    self._validate_conn(conn)
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 1093, in _validate_conn
    conn.connect()
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/connection.py", line 704, in connect
    self.sock = sock = self._new_conn()
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/connection.py", line 207, in _new_conn
    raise ConnectTimeoutError(
urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x7f999aaa9ae0>, 'Connection to murc-kawasesouba.jp timed out. (connect timeout=10.0)')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/runner/work/fx/fx/fx/main.py", line 75, in <module>
    main()
  File "/home/runner/work/fx/fx/fx/main.py", line 71, in main
    args.func(args)
  File "/home/runner/work/fx/fx/fx/update.py", line 54, in update_command
    quotes = download_quotes(provider, base_currency_code, quote_currency_code, start_date, end_date, args.logger)
  File "/home/runner/work/fx/fx/fx/quote.py", line 64, in download_quotes
    quote = provider.get_quote(base_currency_code, quote_currency_code, dt)
  File "/home/runner/work/fx/fx/fx/mufg.py", line 72, in get_quote
    self._cache[(quote_currency_code, quote_date)] = self._get_quotes_by_date(quote_currency_code, quote_date)
  File "/home/runner/work/fx/fx/fx/mufg.py", line 94, in _get_quotes_by_date
    resp = http.request("GET", url)
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/_request_methods.py", line [135](https://github.com/ianlewis/fx/actions/runs/13913162846/job/38931306632#step:5:136), in request
    return self.request_encode_url(
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/_request_methods.py", line 182, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/poolmanager.py", line 443, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 871, in urlopen
    return self.urlopen(
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 871, in urlopen
    return self.urlopen(
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 871, in urlopen
    return self.urlopen(
  [Previous line repeated 2 more times]
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 841, in urlopen
    retries = retries.increment(
  File "/home/runner/work/fx/fx/.venv/lib/python3.10/site-packages/urllib3/util/retry.py", line 519, in increment
    raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='murc-kawasesouba.jp', port=443): Max retries exceeded with url: /fx/past_3month_result.php?y=2025&m=03&d=16 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f999aaa9ae0>, 'Connection to murc-kawasesouba.jp timed out. (connect timeout=10.0)'))
make: *** [Makefile:70: update] Error 1
```

--- a/fx/mufg.py
+++ b/fx/mufg.py
@@ -87,7 +87,8 @@ class MUFGProvider:
         self.logger.debug(f"GET {url}")
 
         http = urllib3.PoolManager(
-            retries=urllib3.Retry(connect=self.retries, read=self.retries, redirect=2),
+            # backoff_factor=0.2 creates retries at 0.0s, 0.4s, 0.8s, 1.6s, ...
+            retries=urllib3.Retry(connect=self.retries, read=self.retries, backoff_factor=0.2, redirect=2),
             timeout=urllib3.Timeout(connect=self.timeout, read=self.timeout),
         )
 
@@ -96,7 +97,9 @@ class MUFGProvider:
         quotes = []
         body = resp.data
         try:
-            # NOTE: Some pages say they are EUC-JP but they are actually SHIFT-JIS.
+            # NOTE: Some pages say they are EUC-JP but they are actually
+            #       SHIFT-JIS so we can't always rely on BeautifulSoup to
+            #       detect the character set.
             body = body.decode("shift-jis")
         except UnicodeDecodeError as e:
             # The decoding error is logged at debug level since it's "expected".

